### PR TITLE
use HEAD pointer in setByPath to increase code readability

### PR DESCRIPTION
**Pull request by malash**
```
      HEAD      path1
root -----> obj -----> ...
             ^
      return root.HEAD
```

### CODE REVIEW

**Comment by ssnau** [comment]
any code sample about the usage?

File: src/utils.js [hunk 1]
```diff
@@ -39,19 +39,19 @@ export const shallowClone = (obj, path = '') => {
   }
 };
 
-export const setByPath = (obj = {}, path = [], value) => {
+const HEAD = 'HEAD';
+export const setByPath = (obj, path = [], value) => {
```
Comment (ssnau) [inline src/utils.js:43]: performance concern: should we check the `obj[path] === value` before calling all the following stuff? If no value changes, you hopefully don't have to anything.

**Comment by malash** [comment]
> any code sample about the usage?

@ssnau  `setByPath` test is here https://github.com/nofluxjs/noflux-state/blob/532ea0c68f0f3eaa36ddbecd355d3434c76518e5/test/utils.js#L49-L61

**Comment by ssnau** [comment]
LGTM. Please consider the performance hit. You can check the value equality before shallow copy every layer on the path.

**Pull request merged**